File: POO/Constructores_Clases/Class/JAVA/Vehiculo.java
public class Vehiculo {
    private String marca;
    private String modelo;
    private int año;
    private double precio;
    private boolean enStock;

    // Constructor con parámetros
    public Vehiculo(String marca, String modelo, int año, double precio, boolean enStock) {
        this.marca = marca;
        this.modelo = modelo;
        this.año = año;
        this.precio = precio;
        this.enStock = enStock;
    }

    public void mostrar() {
        String estado = enStock ? "Disponible" : "No disponible";
        System.out.printf("%s %s (%d) - $%.2f - %s%n", 
            marca, modelo, año, precio, estado);
    }

    public static void main(String[] args) {
        Vehiculo vehiculo = new Vehiculo("Toyota", "Corolla", 2023, 25000.00, true);
        vehiculo.mostrar();
    }
}

File: POO/Constructores_Clases/Class/GO/Vehiculo.go
package main

import "fmt"

type Vehiculo struct {
    marca      string
    modelo     string
    año        int
    precio     float64
    enStock    bool
}

// Constructor con parámetros
func NewVehiculo(marca, modelo string, año int, precio float64, enStock bool) *Vehiculo {
    return &Vehiculo{
        marca:   marca,
        modelo:  modelo,
        año:     año,
        precio:  precio,
        enStock: enStock,
    }
}

func (v *Vehiculo) Mostrar() {
    estado := "No disponible"
    if v.enStock {
        estado = "Disponible"
    }
    fmt.Printf("%s %s (%d) - $%.2f - %s\n", 
        v.marca, v.modelo, v.año, v.precio, estado)
}

func main() {
    vehiculo := NewVehiculo("Toyota", "Corolla", 2023, 25000.00, true)
    vehiculo.Mostrar()
}